www_src/components/modal/modal.jsx

```diff
@@ -0,0 +1,50 @@
+var React = require('react');
+var dispatcher = require('../../lib/dispatcher');
+
+var Modal = React.createClass({
+  getInitialState: function () {
+    return {
+      isVisible: false
+    };
+  },
+  show: function () {
+    this.setState({
+      isVisible: true
+    });
+  },
+  hide: function () {
+    this.setState({
+      isVisible: false
+    });
+  },
+  componentDidMount: function () {
+    dispatcher.on('showModal', (event) => {
```

Comment:
I'm doing something similar for Snackbar notifications (oh god, Android terminology) over at https://github.com/mozilla/webmaker-android/pull/2080 that we might be able to combine (practice wise) for wins on both PRs: if we combine using the dispatcher from your PR with wrapping most of the functionality as a mixin from my PR, it feels like we'd have a powerful "Views that need to house the DOM element that actually achieves notification can do so, and everything that lives *in* a view can just dispatch modal and snackbar messages".

I would advise making the "showModal" a "modal:show" name though (because namespacing), and then handling the event with a separate function like
```
  componentDidMount: function () {
    dispatcher.on('showModal', this.showMessage);
    ...
  },

  showMessage: function(evt) {
    this.setState({
      message: evt.message,
      ...,
      isVisible: true
    });
  }
```